### PR TITLE
Example removal of file-specific PEP8 violation exceptions

### PR DESCRIPTION
**Pull request by dstansby**
Currently pytest.ini has a **lot** of PEP8 exceptions for individual files. This PR demonstrates that with a PEP8 linted text editor it is not that hard to get rid of these exceptions.

### CODE REVIEW

File: lib/matplotlib/backends/qt_editor/formlayout.py [hunk 1]
```diff
@@ -541,14 +543,14 @@ def apply_test(data):
...
-    #--------- datagroup example
+    # datagroup example
     datagroup = create_datagroup_example()
     print("result:", fedit(datagroup, "Global title"))
 
-    #--------- datagroup inside a datagroup example
+    # datagroup inside a datagroup example
     datalist = create_datalist_example()
     datagroup = create_datagroup_example()
     print("result:", fedit(((datagroup, "Title 1", "Tab 1 comment"),
```
Comment (anntzer) [inline lib/matplotlib/backends/qt_editor/formlayout.py]: split print args over three lines

File: lib/matplotlib/backends/qt_editor/formlayout.py [hunk 2]
```diff
@@ -123,7 +120,8 @@ def __init__(self, color, parent=None):
     def update_color(self):
```
Comment (anntzer) [inline lib/matplotlib/backends/qt_editor/formlayout.py]:
```
    def update_color(self):
        # to_qcolor defaults to black for invalid inputs.
        self.colorbtn.color = to_qcolor(self.text())
```
(actually it's not clear at all where the "defaults to black" happens, I just took a random guess)

**Comment by tacaswell** [comment]
This is not false, but dedicated pep8 fixing campaigns have at least once put bugs in, consume a fair amount of review time, and make the git log thrashy.

attn @dopplershift

File: lib/matplotlib/backends/qt_editor/formlayout.py [hunk 3]
```diff
@@ -34,18 +34,12 @@
...
-# 1.0.10: added float validator (disable "Ok" and "Apply" button when not valid)
+# 1.0.10: added float validator
+#   (disable "Ok" and "Apply" button when not valid)
 # 1.0.7: added support for "Apply" button
 # 1.0.6: code cleaning
-
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
-
-__version__ = '1.0.10'
```
Comment (tacaswell) [inline lib/matplotlib/backends/qt_editor/formlayout.py]: these should stay, this is a vendored file.

Comment (tacaswell) [inline lib/matplotlib/backends/qt_editor/formlayout.py]: oh, they were just moved :sheep:

**Comment by dstansby** [comment]
- I think the recent "view previous blame" update to github.com goes some way to mitigating against the problems of PEP8 campaigns (https://github.com/blog/2304-navigate-file-history-faster-with-improved-blame-view)
- I'd argue that given there are strict PEP8 rules for any new pull request, the entire code base should aspire to abide by these rules
- Having these specific exceptions could stop the catching of new PEP8 violations creeping in to the files listed

Am happy to take or leave this though! (which is why I only opened an example)

**Comment by NelleV** [comment]
Some people in this thread may find my comment very ironic, but I think we need to be super careful with those patches. They make rebasing and merging extremely complicated, as they create a lot of changes on different lines.

**Comment by NelleV** [comment]
(Note that I am in favor of a better pep8 testing system, such as the one used on sklearn, that I proposed a couple of months ago. I am happy to reopen the pull request for this one.)

**Comment by dopplershift** [comment]
In general, I'm -1e6 on a gigantic patch pep8-ifying the code. As @tacaswell alluded to, I've had my own code broken by such patches in the past. It's much better, IMO, to pep8 a file, or part of a file, while you work on it.

If we're going to start doing more (small) pep8-only PRs, then at a minimum I think the diff has to have 100% test coverage.

For this particular PR, given that the changes are entirely whitespace and easy to follow, I'm not opposed. One question though: if this is a vendored file, should we be making changes to it at all?

**Comment by NelleV** [comment]
Whitespaces patches are a pain to rebase against.

**Comment by dstansby** [comment]
I'll close this then (esp. since I don't have much experience with version controlling large projects). @NelleV as far as I can tell the sklearn method is to just PEP8 any changes? This would definitely be a good idea I think.

**Comment by NelleV** [comment]
Yes: we have a script that runs on travis to check that all modified lines are pep8. Because it is a local test and not global, it can miss some elements (such as number of blank lines between functions).

**Comment by dopplershift** [comment]
I think in general it might be good to see what files could be taken off the blacklist if only a few lines were changed. Would keep those from acruing such offending lines.

**Comment by QuLogic** [comment]
While what everyone has said here is true about blanket changes, considering you've already done the work and that I can't find any open PRs that affect this file, I don't see why we can't accept _this_ PR specifically.

**Pull request closed without merging**